### DISCUSSION TITLE
In app / automatic updates

### DISCUSSION THREAD

**Discussion by enfJoao** [oqtane/oqtane.framework#4838] [General]
The automatic updates process seems to fail every time, between minor or major versions, but it also does not give any clue to why it happens.

Since stopping the iis site / app pool seems to be a requirement, is there any situation where the automatic updates could work?

**Comment by sbwalker** [comment]
The automatic update works well on IIS deployments. It seems to have issues in Azure (which is why an info message was added to the System Update UI in recent versions). 

The following diagrams outlines the automatic upgrade process:

![image](https://github.com/user-attachments/assets/1c152577-d165-45b8-93ba-7d1c340c646b)

It runs the Updater as an external process which can stop and restart the web process.

**Comment by enfJoao** [comment]
Tried it twice (latest minor update and 6.0 update). Same behavior in both: it downloads the update, allows to click the update button, shows the loading gif and reloads into the same page with no changes. No errors on console or network tab. No errors on server log too.
Using IIS 10 on win11.

![image](https://github.com/user-attachments/assets/19346af1-56db-4575-ae96-9b1e5129a1eb)

Does it rely on any dependency to perform the updates?

**Reply by sbwalker** [reply]
When upgrading to a major Oqtane release (ie. 6.0 ) there are usually other dependencies you need to install in your environment ie. you need to install the latest .NET SDK.

**Reply by sbwalker** [reply]
Note that following the manual upgrade steps provides more control over the process and is recommended in larger installations:

https://www.oqtane.org/blog/!/25/upgrading-oqtane

**Reply by sbwalker** [reply]
Also note that there is a log file created in the /Packages folder specifically for the auto installer which provides details of the activities executed.

**Comment by sbwalker** [comment]
Also... just to confirm... you are NOT trying to use the Auto Upgrade when running Oqtane in Visual Studio - correct?

**Comment by enfJoao** [comment]
No VS. Just IIS, .Net 8 and 9 (during update), SQL Server and the app pool.

Nothing interesting in the package folder.

![image](https://github.com/user-attachments/assets/f2187dde-0692-4e45-8371-4f55b90f1272)

Perhaps the manual update overrode the failed automatic update og.
